### PR TITLE
Add missing properties for Status in the Inspect response

### PR DESCRIPTION
**Pull request by patelrit**
InspectContainerResponse is missing some properties #371

### CODE REVIEW

**Comment by patelrit** [comment]
Added missing Status properties

**Comment by patelrit** [comment]
Any thoughts or comments on this pull request?

Most of these properties have been in the inspect response since api version 1.19

sudo docker version

Client version: 1.7.0

Client API version: 1.19

Go version (client): go1.4.2

Git commit (client): 0baf609

OS/Arch (client): linux/amd64

Server version: 1.7.0

Server API version: 1.19

Go version (server): go1.4.2

Git commit (server): 0baf609

OS/Arch (server): linux/amd64

ubuntu@ip-10-10-129-55:~$ sudo docker inspect d3377a2140e3

[

{

    "Id": "d3377a2140e3150437ade73442633b5689b7aa922f6b953fb5e34013b8de4ea6",

    "Created": "2015-11-18T16:41:19.618214342Z",

    "Path": "catalina.sh",

    "Args": [

        "run"

    ],

    "State": {

        "Running": true,

        "Paused": false,

        "Restarting": false,

        "OOMKilled": false,

        "Dead": false,

        "Pid": 2188,

        "ExitCode": 0,

        "Error": "",

        "StartedAt": "2015-11-18T16:41:19.807255012Z",

**Comment by marcuslinke** [comment]
@patelrit Thanks! Will try to merge it into `2.x` branch ASAP.

**Comment by patelrit** [comment]
Awesome! Thank you!

On Thu, Nov 19, 2015 at 11:39 PM, marcuslinke <notifications@github.com>
wrote:

> @patelrit <https://github.com/patelrit> Thanks! Will try to merge it into
> 2.x branch ASAP.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker-java/docker-java/pull/372#issuecomment-158312691>
> .
>



-- 
Ritesh
Founder @ Nirmata <http://nirmata.com/>

**Comment by KostyaSha** [comment]
Please put `@CheckForNull` everywhere

**Comment by KostyaSha** [comment]
@marcuslinke should this be covered with tests?

**Comment by marcuslinke** [comment]
As we can't test against different docker host versions I think this is not needed. In general a test, if present, should test against the supported docker version of docker-java (as stated in the README)

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java [hunk 1]
```diff
@@ -250,6 +269,18 @@ public Boolean isRunning() {
         public Boolean isPaused() {
             return paused;
         }
+        
+        public Boolean isRestarting() {
+            return restarting;
+        }
+        
+        public Boolean isOOMKilled() {
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Boolean shouldn't be used with `is` as it not a primitive, use `get`

**Comment by KostyaSha** [comment]
> As we can't test against different docker host versions I think this is not needed

Bad...

**Comment by KostyaSha** [comment]
I will handle this PR and sync into 2.3 and master.

**Comment by marcuslinke** [comment]
@KostyaSha :+1:

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java [hunk 2]
```diff
@@ -225,17 +225,32 @@ public String toString() {
...
         private Boolean paused;
 
+        @JsonProperty("Restarting")
+        private Boolean restarting;
+        
+        @JsonProperty("OOMKilled")
+        private Boolean oomKilled;
+        
+        @JsonProperty("Dead")
+        private Boolean dead;
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Didn't found in 1.17..1.21 please point on API documentation

Comment (patelrit) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Please see:
https://docs.docker.com/v1.8/reference/api/docker_remote_api_v1.17/
"State": { *"Error": ""**, * "ExitCode": 9, "FinishedAt":
"2015-01-06T15:47:32.080254511Z", *"OOMKilled": false**, * "Paused": false,
"Pid": 0, *"Restarting": false**, * "Running": false, "StartedAt":
"2015-01-06T15:47:32.072697474Z" },

Also, while the docker document does not have it, I see some additional
properties in the latest 1.21 (Docker 1.9 API). I can remove them if you
strictly want to go by the documentation but usually docker documentation
is always behind..

    "State": {

*        "Status": "exited",*

        "Running": false,

        "Paused": false,

        "Restarting": false,

        "OOMKilled": false,

*        "Dead": false,*

        "Pid": 0,

        "ExitCode": 0,

        "Error": "",

        "StartedAt": "2015-11-20T19:02:48.972024978Z",

        "FinishedAt": "2015-11-20T19:02:48.998958231Z"

    }


On Fri, Nov 20, 2015 at 10:54 AM, Kanstantsin Shautsou <
notifications@github.com> wrote:

> In
> src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java
> <https://github.com/docker-java/docker-java/pull/372#discussion_r45503748>
> :
>
> >          @JsonProperty("Running")
> >          private Boolean running;
> >
> >          @JsonProperty("Paused")
> >          private Boolean paused;
> >
> > +        @JsonProperty("Restarting")
> > +        private Boolean restarting;
> > +
> > +        @JsonProperty("OOMKilled")
> > +        private Boolean oomKilled;
> > +
> > +        @JsonProperty("Dead")
> > +        private Boolean dead;
>
> Didn't found in 1.17..1.21 please point on API documentation
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker-java/docker-java/pull/372/files#r45503748>.
>



-- 
Ritesh
Founder @ Nirmata <http://nirmata.com/>

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java (hunk 1 above)
Comment (patelrit) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Just trying to be consistent with the existing methods isRunning & isPaused.

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java (hunk 2 above)
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Filled https://github.com/docker/docker/pull/18127

**Comment by patelrit** [comment]
@CheckForNull everywhere

**Comment by KostyaSha** [comment]
@patelrit please hold on, i already reworked all related changes.

**Comment by KostyaSha** [comment]
@patelrit #378 :atm: but i continue implementing tests.

File: src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java (hunk 2 above)
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/api/command/InspectContainerResponse.java]: Upstream merged

**Comment by KostyaSha** [comment]
I picked your change and modified in #378 closing this.

**Pull request closed without merging**